lib/config/openssh/openssh.go

```diff
@@ -35,7 +35,18 @@ import (
 	"github.com/gravitational/teleport/lib/utils"
 )
 
-var sshConfigTemplate = template.Must(template.New("ssh-config").Parse(
+var sshConfigTemplate = template.Must(template.New("ssh-config").Funcs(template.FuncMap{
+	// shellQuote prepares a string for insertion into the ssh_config
+	"shellQuote": func(s string) string {
+		s = `'` + strings.ReplaceAll(s, `'`, `'"'"'`) + `'`
+		// escape any percent signs which could trigger the percent expansion
+		// for ProxyCommand.
+		s = strings.ReplaceAll(s, `%`, `%%`)
+		// escape any newlines which could impact the parsing of ssh_config
+		s = strings.ReplaceAll(s, "\n", `'"\n"'`)
```

Comment:
We couldn't find a sensible way to then go back and replace literal newlines with their escaped form with a `shsprintf.DefaultWhatever` - which we need because we're trying to build a string that, when fed through openssh's config parsing, percent replacing and execution through a shell, results in the process we want being executed with the arguments that we want.